exchanges/okx/okx_test.go

```diff
@@ -3369,372 +2930,951 @@ func TestGetHistoricalFundingRates(t *testing.T) {
 		r.IncludePayments = true
 	}
 	_, err = ok.GetHistoricalFundingRates(contextGenerate(), r)
-	if err != nil {
-		t.Error(err)
-	}
+	assert.NoError(t, err)
 
 	r.StartDate = time.Now().Add(-time.Hour * 24 * 120)
 	_, err = ok.GetHistoricalFundingRates(contextGenerate(), r)
-	if !errors.Is(err, fundingrate.ErrFundingRateOutsideLimits) {
-		t.Error(err)
-	}
+	assert.True(t, errors.Is(err, fundingrate.ErrFundingRateOutsideLimits), err)
 
 	r.RespectHistoryLimits = true
 	_, err = ok.GetHistoricalFundingRates(contextGenerate(), r)
-	if err != nil {
-		t.Error(err)
-	}
+	assert.NoError(t, err)
 }
 
 func TestIsPerpetualFutureCurrency(t *testing.T) {
 	t.Parallel()
 	is, err := ok.IsPerpetualFutureCurrency(asset.Binary, currency.NewPair(currency.BTC, currency.USDT))
-	if err != nil {
-		t.Error(err)
-	}
-	if is {
-		t.Error("expected false")
-	}
+	assert.NoError(t, err)
+	assert.False(t, is, "expected false")
 
 	cp, err := currency.NewPairFromString("BTC-USD-SWAP")
-	if err != nil {
-		t.Error(err)
-	}
+	assert.NoError(t, err)
 	is, err = ok.IsPerpetualFutureCurrency(asset.PerpetualSwap, cp)
-	if err != nil {
-		t.Error(err)
-	}
-	if !is {
-		t.Error("expected true")
-	}
+	assert.NoError(t, err)
+	assert.True(t, is, "expected true")
 }
 
 func TestGetAssetsFromInstrumentTypeOrID(t *testing.T) {
 	t.Parallel()
 	_, err := ok.GetAssetsFromInstrumentTypeOrID("", "")
-	if !errors.Is(err, errEmptyArgument) {
-		t.Error(err)
-	}
+	assert.True(t, errors.Is(err, errEmptyArgument), err)
 
 	assets, err := ok.GetAssetsFromInstrumentTypeOrID("SPOT", "")
-	if !errors.Is(err, nil) {
-		t.Error(err)
-	}
-	if len(assets) != 1 {
-		t.Errorf("received %v expected %v", len(assets), 1)
-	}
-	if assets[0] != asset.Spot {
-		t.Errorf("received %v expected %v", assets[0], asset.Spot)
-	}
+	assert.NoError(t, err)
+	assert.Falsef(t, len(assets) != 1, "received %v expected %v", len(assets), 1)
+	assert.Falsef(t, assets[0] != asset.Spot, "received %v expected %v", assets[0], asset.Spot)
 
 	assets, err = ok.GetAssetsFromInstrumentTypeOrID("", ok.CurrencyPairs.Pairs[asset.Futures].Enabled[0].String())
-	if !errors.Is(err, nil) {
-		t.Error(err)
-	}
-	if len(assets) != 1 {
-		t.Errorf("received %v expected %v", len(assets), 1)
-	}
-	if assets[0] != asset.Futures {
-		t.Errorf("received %v expected %v", assets[0], asset.Futures)
-	}
+	assert.NoError(t, err)
```

Comment:
use package require `require.NoError(t, err)` as it can panic below on failure and across this file